.github/workflows/sphinx.yml

```
@@ -32,10 +37,25 @@ jobs:
           SYSTEM_PYTHON: python${{ matrix.python-version }}
           SECRET_KEY: github-actions
 
-      - name: Deploy to GitHub Pages
-        if: ${{ github.ref == 'refs/heads/main' }}
-        uses: JamesIves/github-pages-deploy-action@v4.4.1
+      - name: Upload pages artifact
+        uses: actions/upload-pages-artifact@v2
         with:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          BRANCH: gh-pages
-          FOLDER: dist/html
+          path: 'dist/html'
+
+  deploy:
+    needs: build
+    if: ${{ github.ref == 'refs/heads/master' }}
```

I wonder if it would be better to use the variable for this 🤔 
```suggestion
    if: ${{ github.ref == github.event.repository.default_branch }}
```

---

I think we need to compare with `refs/heads/{default_branch}`

(https://stackoverflow.com/a/69995861/4295499)

---

Ah, I guess there is also `github.ref_name` that could work in this case.

```suggestion
    if: ${{ github.ref_name == github.event.repository.default_branch }}
```

Could work. Your solution also works, but I think I sometimes over think trying to find the simplest solution 😆

---

@kfdm gentle ping :)

---

less than 24 hours is a bit soon for a ping. Sometimes other work tasks get in the way of checking pull requests 😅

---

haha sorry 😅